morpho/processors/diagnostics/CalibrationProcessor.py

```diff
@@ -0,0 +1,278 @@
+'''
+Processor for calibrating results; i.e., determining how often posteriors are consistent with "true" values assumed to generate fake data.
+Useful for sensitivity analyses.
+
+Authors: T. E. Weiss
+Date: May 2020
+'''
+
+from __future__ import absolute_import
+
+import numpy as np
+import math
+
+from morpho.utilities import morphologging, reader
+from morpho.processors import BaseProcessor
+from morpho.processors.IO import IOROOTProcessor
+logger = morphologging.getLogger(__name__)
+
+__all__ = []
+__all__.append(__name__)
+
+
+class CalibrationProcessor(BaseProcessor):
+    '''
+    Performs a Bayesian sensitivity calibration for a continuous parameter - i.e., computes the coverage of a credible interval. Uses either an upper limit or upper and lower bounds on a posterior, depending on user input. Prints the coverage as well as (optionally) the median and mean credible windows.
+    
+    Required input:
+        files: List of strings naming ROOT files produced by an ensemble of morpho runs.
+        in_param_names: List of strings naming parameters of interest inputted to the generator.
+        
+    Optional input:
+        cred_interval: List with float elements between 0 and 1 defining a posterior credible window; defaults to [0.05, 0.95]. If len(cred_interval)==1, this procesor finds the coverage of a limit. If len(cred_interval)==2, it finds the coverage of an interval.
+        root_in_tree: Tree containing data generation input values in each file. Defaults to "input".
+        root_post_tree: Tree containing analysis posteriors. Defaults to "analysis".
+        post_param_names: List of strings naming posteriors produced by Stan analysis for a parameters of interest. Defaults to self.in_param_names.
+        verbose: If True, print information for each pseudo-experiment and a summary at the end. If False, only print the summary.
```

Comment:
I needed to modify the morphologging to be able to implement my comment... 😅

Reply:
Great, and sorry that modifying morphologging took a while!

Reply:
I should say that I am okay with using logger.info here instead of logger.debug. Should we still change that?

Reply:
no worries! it needed to be done.
Now, it is possible to have a common logging system for both processors and analysis scripts.
To do so, you would only need to add 
```python
    args = parser.parse_args(False)
    logger = morphologging.getLogger('morpho',
                                     level=args.verbosity,
                                     stderr_lb=args.stderr_verbosity,
                                     propagate=False)
```
in your main() function and changing the verbosity level using `-v` or `-vv`.
An example can be found in https://github.com/morphoorg/morpho/blob/master/tests/misc/script_logger.py

No `-v` will only print warning and error messages.
1 `-v` will print warning, error and info messages.
2 `-v`'s or 1 `-vv` (or more) will print every kind of messages (including debug).
Of course the default level of verbosity can be changed: if you have inputs or preferences, we can for sure change this...

Reply:
Thank you for the explanation and example script.

Do I add those two lines in scripts that are being used to run CalibrationProcessor?

I think that using this for the CalibrationProcessor will work well if the default can be `-v `(warning, error and info messages), and then people can add more `v`s if desired. Can I set the default verbosity in InternalConfigure of the CalibrationProcessor?

Reply:
Yes, you can: this would be in the final-analysis script in mermithid right?

OK, I can change the default to be warning, error and info.
I will do this in the feature/calib_fixes branch (it sounds to me like a very minor change)...